src/components/Snippet/FileManager.vue

```diff
@@ -0,0 +1,215 @@
+<script>
+import { setTimeout } from 'timers';
+
+import {
+  Upload as GirderUpload,
+  UpsertFolder as GirderUpsertFolder,
+  DataBrowser as GirderDataBrowser,
+  Breadcrumb as GirderBreadcrumb,
+} from '../';
+import {
+  getLocationType,
+  isRootLocation,
+  createLocationValidator,
+} from '../../utils';
+
+export default {
+  components: {
+    GirderBreadcrumb,
+    GirderUpload,
+    GirderUpsertFolder,
+    GirderDataBrowser,
+  },
+
+  props: {
+    location: {
+      type: Object,
+      validator: createLocationValidator(true),
+      default: () => { 'root'; },
+    },
+    rootLocationAllowed: {
+      type: Boolean,
+      default: false,
+    },
+    selectable: {
+      type: Boolean,
+      default: false,
+    },
+    dragEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    uploadEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    newFolderEnabled: {
+      type: Boolean,
+      default: false,
+    },
+    uploadMultiple: {
+      type: Boolean,
+      default: false,
+    },
+    uploadAccept: {
+      type: String,
+      default: '*',
+    },
+    preUpload: {
+      type: Function,
+      default: async () => {},
+    },
+    postUpload: {
+      type: Function,
+      default: async () => {},
+    },
+    preUpsert: {
+      type: Function,
+      default: async () => {},
+    },
+    postUpsert: {
+      type: Function,
+      default: async () => {},
+    },
+  },
+
+  inject: ['girderRest'],
+
+  data() {
+    return {
+      uploaderDialog: false,
+      newFolderDialog: false,
+      lazyLocation: null,
+    };
+  },
+
+  asyncComputed: {
+    async defaultLocation() {
```

Comment:
The default upload destination behavior is confusing. When a user is on its user root folder, then upload a file, the file won't be showing up. I think we should simply only allow uploading to the current folder.

Reply:
So then should we disable the Upload button when the user is in their user root?

Reply:
Yes, I had a computed to check if the current location is a folder. So, even if upload is enabled, but the current location is not a folder, the button won't show.